pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,316 @@
+
+By calling a route decorator, you provide OpenAPI specification to describe the 
+endpoint which the decorated method maps to. You can choose different decorators 
+accordingly or do a composition of them:
+
+### API Decorator
+
+  Syntax: `@api`
+
+  `@api` is a decorator for controller constructor, it's called before a controller
+  class. `@api` is used when you have a base path and a Paths Object, which
+  contains all path definitions of your controller. Please note the api specs defined 
+  with `@api` will override other api specs defined inside the controller. For example:
+
+  ```ts
+    @api({
+      basePath: '/',
+      paths: {
+        '/greet': {
+          get: {
+            'x-operation-name': 'greet',
+            'x-controller-name': 'MyController',
+            parameters: [{name: 'name', type: 'string', in: 'query'}],
+            responses: {
+              '200': {
+                description: 'greeting text',
+                schema: {type: 'string'},
+              }
+            }
+          }
+        }
+      }
+    })
+    class MyController {
+      // The operation endpoint defined here will be overriden!
+      @get('/foo')
+      @param.query.number('limit')
+      greet(name) {
+      }
+    }
+    app.controller(MyController);
+  ```
+
+  A more detailed explanation can be found in [Specifying Controller APIs](controllers.htm#specifying-controller-apis)
+
+### Operation Decorator
+
+  Syntax: `@operation`
+
+  `@operation` is a controller method decorator. It exposes a Controller method as 
+  a REST API operation. You can specify the verb, path, parameters and response 
+  as specification of your endpoint, for example:
+
+  ```ts
+  const spec = {
+    parameters: [{name: 'name', type: 'string', in: 'query'}],
+    responses: {
+      '200': {
+        description: 'greeting text',
+        schema: {type: 'boolean'},
+      }
+    }
+  };
+  class MyController {
+    @operation('HEAD', '/checkExist', spec)
+    checkExist(name) {
+    }
+  }
+  ```
+  
+### Commonly-used Operations Decorator
+  
+  Syntax: `@get`, `@post`, `@put`, `@patch`, `@del`
+
+  You can call these suger operation decorators as a shortcut of `@operation`, for example:
+  ```ts
+  class MyController {
+    @get('/greet', spec)
+    greet(name) {
+    }
+  }
+  ```
+
+  is equivalent to
+
+  ```ts
+  class MyController {
+    @operation('GET', '/greet', spec)
+    greet(name) {
+    }
+  }
+  ```
+  
+  For more usage, refer to [Routing to Controllers](controllers.htm#routing-to-controllers)
+
+### Parameter Decorator
+
+  Syntax: `@param`
+
+  Optional Pattern: `@param.${in}.${type}(${name})`
+
+  `@param` can be applied to method itself or specific parameters, it describes
+  an input parameter of a Controller method.
+
+  For example:
+
+  ```ts
+    class MyController {
+      @get('/')
+      @param(offsetSpec)
+      @param(pageSizeSpec)
+      list(offset?: number, pageSize?: number) {}
```

Comment:
Add a note that method level `@param` is positional and it's exclusive with parameter level `@param`

Reply:
Can you expand on what that means? Personally, I _think_ I know what you mean, but I personally wouldn't act on the feedback if I didn't know what it meant.

Reply:
Let me confirm I understand it correctly... "positional" means the sequence of composed decorator must be
```
@operationDecorator
@param
```

> exclusive with parameter level @param

Means "Mixed usage of @param at method/parameter level is not allowed and results in error"

Reply:
@raymondfeng We really need an answer here.

Reply:
@kjdelisle See https://github.com/strongloop/loopback-next/blob/master/packages/rest/test/unit/router/metadata/param.test.ts#L40-L119